iOS/Sources/Beagle/Sources/Components/Layout+Renderable/Screen/Tests/ScreenComponentTests.swift

```diff
@@ -92,6 +92,32 @@ final class ScreenComponentTests: XCTestCase {
         assertSnapshotImage(viewController, size: .custom(CGSize(width: 300, height: 200)))
     }
     
+    func testNavigationBarItemWithContextOnImage() {
+        //Given
+        let dependencies = BeagleDependencies()
+        dependencies.appBundle = Bundle(for: ScreenComponentTests.self)
+        
+        let barItem = NavigationBarItem(image: "@{image}", text: "", action: ActionDummy())
+        let component = Screen(
+            safeArea: SafeArea.all,
+            navigationBar: .init(title: "title", showBackButton: true, navigationBarItems: [barItem]),
+            child: Text("test")
+        )
+
+        let controller = BeagleScreenViewController(viewModel: .init(screenType:.declarative(component), dependencies: dependencies))
+        let action = SetContext(contextId: "image", value: "shuttle")
+        let view = component.toView(renderer: controller.renderer)
+        controller.content = .view(view)
+        
+        //When
+        controller.view.setContext(Context(id: "image", value: "test_image_square-x"))
+        controller.configBindings()
+        action.execute(controller: controller, origin: controller.view)
+        
+        // Then
+        assertSnapshotImage(controller.view, size: ImageSize.custom(CGSize(width: 300, height: 200)))
+    }
```

Comment:
I think that if we only test the first time of `observe` being called, we should already know that it works. This change will allow us reduce weird code, and we can leave the check for multiple `observe` iteration to more specific `observe` tests

```suggestion
    func testNavigationBarItemWithContextOnImage() {
        // Given
        let dependencies = BeagleDependencies()
        dependencies.appBundle = Bundle(for: ScreenComponentTests.self)
        
        let barItem = NavigationBarItem(image: "@{image}", text: "", action: ActionDummy())

        let screen = Screen(
            safeArea: SafeArea.all,
            navigationBar: .init(title: "title", showBackButton: true, navigationBarItems: [barItem]),
            child: Text("test"),
            context: Context(id: "image", value: "shuttle")
        )

        let controller = BeagleScreenViewController(viewModel: .init(
            screenType: .declarative(screen),
            dependencies: dependencies
        ))
        
        // Then
        assertSnapshotImage(controller, size: .custom(CGSize(width: 150, height: 80)))
    }
```

Reply:
nice, that is much better!